accounts/views.py

```diff
@@ -21,33 +24,40 @@ def register(request):
-
-            user = Account.objects.create_user(
-                                first_name=first_name,
-                                last_name=last_name,
-                                email=email,
-                                password=password,
-                                username=username
-                                )
-            user.phone_number = phone_number
-            user.save()
-
-            current_site = get_current_site(request)
-            mail_subject = "Please activate your account!"
-            message = render_to_string('accounts/account_verification_email.html', {
-                'user': user,
-                'domain': current_site,
-                'uid': urlsafe_base64_encode(force_bytes(user.id)),
-                'token': default_token_generator.make_token(user),
-            })
-            send_email = EmailMessage(mail_subject, message, to=[email])
-            send_email.send()
+            try:
+                with transaction.atomic():
+                    first_name = form.cleaned_data['first_name']
+                    last_name = form.cleaned_data['last_name']
+                    email = form.cleaned_data['email']
+                    phone_number = form.cleaned_data['phone_number']
+                    password = form.cleaned_data['password']
+                    username = email.split('@')[0]
+
+                    user = Account.objects.create_user(
+                                        first_name=first_name,
+                                        last_name=last_name,
+                                        email=email,
+                                        password=password,
+                                        username=username
+                                        )
+                    user.phone_number = phone_number
+                    user.save()
+
+                    current_site = get_current_site(request)
+                    mail_subject = "Please activate your account!"
+                    message = render_to_string('accounts/account_verification_email.html', {
+                        'user': user,
+                        'domain': current_site,
+                        'uid': urlsafe_base64_encode(force_bytes(user.id)),
+                        'token': default_token_generator.make_token(user),
+                    })
+                    send_email = EmailMessage(mail_subject, message, to=[email])
+                    send_email.send()
+            except Exception as e:
+                    transaction.set_rollback(True)
+                    messages.error(request, 'An error occurred during registration.')
+                    return redirect('register')
+            
             messages.success(request, 'Thank you for registering with us. We have s sent email verification to your email address. Please verify it.')
             return redirect('/accounts/login/?command=verification&email='+email)
     else:
```

Comment:
this is not clean code:

```
if request.method == "POST":
        form = RegistrationForm(request.POST)
        if form.is_valid():
                .....
else:
      form = RegistrationForm()
```

Why we need an explicit `else` statement?

```
form = RegistrationForm()

if request.method == "POST":
        form = RegistrationForm(request.POST)
        if form.is_valid():
                .....
```

this is a bit cleaner because if the request method is not post, then it defaults to the initial form, MAZGI!